sssd_test_framework/utils/pam.py

```diff
@@ -0,0 +1,284 @@
+
+        self.fs: LinuxFileSystem = fs
+
+    def access(self, file: str = "/etc/security/access.conf") -> PAMAccess:
+        """
+        :param file: PAM Access file name.
+        :type file: str
+        :return: PAM Access object
+        :rtype: PAMAccess
+        """
+        return PAMAccess(self, file)
+
+    def faillock(self, file: str = "/etc/security/faillock.conf") -> PAMFaillock:
+        """
+        :param file: PAM Faillock file name.
+        :type file: str
+        :return: PAM Faillock object
+        :rtype: PAMFaillock
+        """
+        return PAMFaillock(self, file)
+
+
+class PAMAccess:
+    """
+    Management of PAM Access on the client host.
+
+    .. code-block:: python
+        :caption: Example usage
+
+        @pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+        def test_example(client: Client, provider: GenericProvider):
+            # Add users
+            provider.user("user-1").add()
+            provider.user("user-2").add()
+
+            # Add rule to permit "user-1" and deny "user-2"
+            access = client.pam.access()
+            access.config_set([{"access": "+", "user": "user-1", "origin": "ALL"},
+                               {"access": "-", "user": "user-2", "origin": "ALL"}])
+            access.config_write()
+
+            client.sssd.authselect.enable_feature(["with-pamaccess"])
+            client.sssd.start()
+
+            # Check the results
+            assert client.auth.ssh.password("user-1", "Secret123")
+            assert not client.auth.ssh.password("user-2", "Secret123")
+    """
+
+    def __init__(self, util: PAMUtils, file: str = "/etc/security/access.conf") -> None:
+        """
+        :param util: PAMUtils utility object
+        :type util: PAMUtils
+        :param file: File name of access file
+        :type file: str, optional
+        """
+        self.util: PAMUtils = util
+        self.file: str = file
+        self.path: str = "/files" + self.file
+        self.args: str = f'--noautoload --transform "Access.lns incl {self.file}"'
+        self.cmd: str = ""
+
+    def config_write(self):
+        """
+        Write access configuration.
+        :return: None
+        """
+        self.util.fs.backup(self.file)
+        self.util.logger.info(f"Saving Augeas changes to {self.file} on {self.util.host.hostname}")
+        self.util.host.ssh.run(f'echo -e "{self.cmd} save\n" | augtool --echo {self.args}')
+
+    def config_read(self) -> str:
+        """
+        Read access file as Augeas tree.
+        :return: PAM access configuration
+        :rtype: str
+        """
+        self.util.logger.info(f"Reading Augeas tree {self.file} on {self.util.host.hostname}")
+        result = self.util.host.ssh.run(f"augtool {self.args} print {self.path}")
+
+        return result.stdout
+
+    def config_delete(self, value: dict[str, str]) -> None:
+        """
+        Delete access configuration.
+
+        TODO: Needs testing
```

Comment:
You forgot this `TODO`, or maybe you didn't and you need to test it. In any case, I'm skipping this function review until this point is clarified.

Reply:
This function will only be written because it makes sense to have it. I don't know if we need to add it, since none of the test code uses it. I'm unsure if we just want to remove it and add it later if there is a need for it?

Reply:
Unless you have a use case already in mind where this would be useful I'd remove it. We can always add it when we find a use case.

Reply:
We discussed it, it will be a good reference for other augeas delete functions and I'm almost done with it.